### PR TITLE
add parameters as keyword arguments

### PR DESCRIPTION
**Pull request by MonssafToukal**

### CODE REVIEW

**Review by tmigot — CHANGES_REQUESTED** [review]
Looks great, thank you! I just suggested two modifications.

File: src/lbfgs.jl [hunk 1]
```diff
@@ -18,8 +18,10 @@ The keyword arguments may include
 - `rtol::T = √eps(T)`: relative tolerance, the algorithm stops when ||∇f(xᵏ)|| ≤ atol + rtol * ||∇f(x⁰)||.
 - `max_eval::Int = -1`: maximum number of objective function evaluations.
 - `max_time::Float64 = 30.0`: maximum time limit in seconds.
+- `τ₁::T = T(0.9999)`: slope factor in the Wolfe condition when performing the line search.
+- `bk_max:: Int = 25`: maximum number of backtracks when performing the line search.
 - `verbose::Int = 0`: if > 0, display iteration details every `verbose` iteration.
-- `mem::Int = 5`: memory parameter of the `lbfgs` algorithm.
```
Comment (tmigot) [inline src/lbfgs.jl:22]: You can keep `mem` in the docstring as it is used by the function `lbfgs`.

Oh, I actually realize that it's not and this is a bug
https://github.com/ProofOfConceptForJuliSmoothOptimizers/JSOSolvers.jl/blob/691905208b88a7e1c30ef8027ccf5f4922ab3190/src/lbfgs.jl#L84
should have `mem` in the `kwargs` of the function `lbfgs` and pass it to the `LBFGSSolver`

Comment (MonssafToukal) [reply src/lbfgs.jl:22]: Oh I see! No problem, I will fix it.

**Review by tmigot — APPROVED** [review]
Thanks!

File: src/lbfgs.jl [hunk 2]
```diff
@@ -146,7 +152,7 @@ function solve!(
 
     # Perform improved Armijo linesearch.
     t, good_grad, ft, nbk, nbW =
-      armijo_wolfe(h, f, slope, ∇ft, τ₁ = T(0.9999), bk_max = 25, verbose = false)
+      armijo_wolfe(h, f, slope, ∇ft, τ₁ = τ₁, bk_max = bk_max, verbose = Bool(verbose_subsolver))
```
Comment (dpo) [inline src/lbfgs.jl:155]: Why is `verbose` and int converted to bool?

Comment (tmigot) [reply src/lbfgs.jl:155]: I think it is to be consistent with `trunk` https://github.com/JuliaSmoothOptimizers/JSOSolvers.jl/blob/f91f34d4eb374323ea582838a2747d2ddeaa2a09/src/trunk.jl#L129

**Pull request merged**